Add io.github.smal82.LinuxMagnetDistro

smal82 opened the pull request:
<!-- ⚠️⚠️ Submission pull request MUST be made against the `new-pr` **base branch** ⚠️⚠️  -->

<!-- 💡 Go to the preview tab to click the links below 💡 -->

### Please confirm your submission meets all the criteria

<!-- 💡 Please replace each `[ ]` with `[X]` when the step is complete 💡 -->

Linux Magnet Distro GNOME is a modern desktop app, built in Python+GTK, that serves as a central hub for discovering, downloading, and managing ISOs for major Linux distributions.
    The goal is to make it quick and easy to try new distros, while staying up to date with the latest news in the Linux universe!
    The domain used for the application ID (io.github.smal82.LinuxMagnetDistro) is controlled by me, the application developer, and follows the application ID guidelines.
    I have read and followed all the Submission requirements and the Submission guide, and I agree to them.
    I am the author and upstream developer of the project.

<!-- 💡 Mention the GitHub usernames of any additional maintainers needed below 💡 -->

<!-- ⚠️⚠️ DO NOT modify anything below this line ⚠️⚠️  -->

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[reqs2]: https://docs.flathub.org/docs/for-app-authors/submission

bbhtt commented:
If you remove the template then we can close this...

bbhtt commented on io.github.smal82.LinuxMagnetDistro/io.github.smal82.LinuxMagnetDistro.yml:
```diff
@@ -0,0 +1,40 @@
+sdk: org.gnome.Sdk
+command: linux-magnet-distro
+# Opzionale: metadata
+title: Linux Magnet Distro
+description: "L'hub per ISO Linux. Scopri, scarica e vivi la libertà del software open source."
+homepage: "https://github.com/smal82/Linux-Magnet-Distro-GNOME"
```
these must be written by AI...

smal82 commented:
Thank you for your feedback!
I have now removed the PR template file and updated the manifest and desktop file according to your suggestions.
Please let me know if anything else needs to be changed.

hfiguiere reviewed (changes requested):
This doesn't follow the required layout for the file in the submission. Nothing can be done until it's addressed.

hfiguiere commented on io.github.smal82.LinuxMagnetDistro/io.github.smal82.LinuxMagnetDistro.desktop:
```diff
@@ -0,0 +1,11 @@
+Type=Application
+Categories=Utility;
+StartupNotify=true
+# Optional
+Homepage=https://github.com/smal82/Linux-Magnet-Distro-GNOME
```
I don't see where this is an existing key https://specifications.freedesktop.org/desktop-entry-spec/1.5/recognized-keys.html

hfiguiere replied:
```suggestion
```

hfiguiere commented on io.github.smal82.LinuxMagnetDistro/io.github.smal82.LinuxMagnetDistro.desktop:
```diff
@@ -0,0 +1,11 @@
+[Desktop Entry]
+Name=Linux Magnet Distro
+Comment=L'hub per ISO Linux. Scopri, scarica e vivi la libertà del software open source.
```
this should be in English. you can add a localized version.

hfiguiere commented on io.github.smal82.LinuxMagnetDistro/io.github.smal82.LinuxMagnetDistro.yml:
```diff
@@ -0,0 +1,36 @@
+app-id: io.github.smal82.LinuxMagnetDistro
+runtime: org.gnome.Platform
+runtime-version: "46"
```
46 is EOL. 48 is the current one.

hfiguiere commented on io.github.smal82.LinuxMagnetDistro/io.github.smal82.LinuxMagnetDistro.yml:
```diff
@@ -0,0 +1,36 @@
+      - pip3 install --prefix=/app sgmllib3k-1.0.0.tar.gz
+    sources:
+      - type: file
+        path: third-party/sgmllib3k-1.0.0.tar.gz
```
this file doesn't exist.
(and adding it is NOT the solution either.

hfiguiere commented on io.github.smal82.LinuxMagnetDistro/io.github.smal82.LinuxMagnetDistro.yml:
```diff
@@ -0,0 +1,36 @@
+      - pip3 install --prefix=/app feedparser-6.0.11.tar.gz
+    sources:
+      - type: file
+        path: third-party/feedparser-6.0.11.tar.gz
```
this file doesn't exist.
(and adding it is NOT the solution either.

hfiguiere replied:
You did the opposite of what I wrote above.

hfiguiere commented on io.github.smal82.LinuxMagnetDistro/io.github.smal82.LinuxMagnetDistro.yml:
```diff
@@ -0,0 +1,36 @@
+      - install -D theme.css ${FLATPAK_DEST}/share/linux-magnet-distro/theme.css
+      - install -D logo.png ${FLATPAK_DEST}/share/linux-magnet-distro/logo.png
+      - install -D io.github.smal82.LinuxMagnetDistro.desktop ${FLATPAK_DEST}/share/applications/io.github.smal82.LinuxMagnetDistro.desktop
+      - install -D io.github.smal82.LinuxMagnetDistro.png ${FLATPAK_DEST}/share/icons/hicolor/128x128/apps/io.github.smal82.LinuxMagnetDistro.png
```
this icon file is NOT 128x128.

hfiguiere commented on io.github.smal82.LinuxMagnetDistro/io.github.smal82.LinuxMagnetDistro.yml:
```diff
@@ -0,0 +1,36 @@
+      - install -D io.github.smal82.LinuxMagnetDistro.desktop ${FLATPAK_DEST}/share/applications/io.github.smal82.LinuxMagnetDistro.desktop
+      - install -D io.github.smal82.LinuxMagnetDistro.png ${FLATPAK_DEST}/share/icons/hicolor/128x128/apps/io.github.smal82.LinuxMagnetDistro.png
+    sources:
+      - type: dir
```
don't use source `type: dir`.

hfiguiere replied:
also there is no way for flatpak-builder to know where to get the source.

hfiguiere commented on io.github.smal82.LinuxMagnetDistro/io.github.smal82.LinuxMagnetDistro.yml:
```diff
@@ -0,0 +1,36 @@
+    sources:
+      - type: dir
+        path: .
+finish-args:
```
`finish-args` should be above `modules`

hfiguiere commented on io.github.smal82.LinuxMagnetDistro/io.github.smal82.LinuxMagnetDistro.yml:
```diff
@@ -0,0 +1,36 @@
+        path: .
+finish-args:
+  - --share=network
+  - --filesystem=home
```
why does it need home access.

hfiguiere commented on io.github.smal82.LinuxMagnetDistro/io.github.smal82.LinuxMagnetDistro.yml:
```diff
@@ -0,0 +1,36 @@
+  - --share=network
+  - --filesystem=home
+  - --socket=fallback-x11
+  - --socket=wayland
```
`--share=ipc` is needed too.

hfiguiere reviewed (changes requested):
You merged master into this. Not sure why you would do that, but this need to be reverted.

Also you ignored the part where IO said the python file should never be added to here.

smal82 commented:
Hi,
I can't solve the reported problems. So I apologize for wasting your time. You can close and delete this PR.

Good work.
